crates/typst/src/lib.rs

```diff
@@ -231,6 +231,51 @@ pub trait World {
     }
 }
 
+macro_rules! delegate_for_ptr {
+    ($W:ident for $ptr:ty) => {
+        impl<$W: World> World for $ptr
+        where
+            $ptr: Deref<Target = W>,
```

Comment:
Probably this should be

```suggestion
            $ptr: Deref<Target = $W>,
```

Reply:
Indeed, a small oversight, I wonder how this nonetheless compiles, I was under the impression that macro hygiene works both ways. Maybe it doesn't apply to generics of `$ty` fragments.